### PR TITLE
Add support of NuoDB into querydsl-sql

### PR DESCRIPTION
**Pull request by tazija**
Introduced com.querydsl.sql.NuoDBTemplates
Added com.querydsl.sql.nuodb.NuoDBQuery
com.querydsl.sql.suites.NuoDBSuiteTest verified to run: Tests run: 280, Failures: 0, Errors: 0, Skipped: 4

### CODE REVIEW

File: querydsl-jpa/pom.xml [hunk 1]
```diff
@@ -182,6 +182,12 @@
       <version>${mysql.version}</version>
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>mysql</groupId>
```
Comment (timowest) [inline querydsl-jpa/pom.xml]: why the duplicate mysql dependency?

File: querydsl-sql/src/test/java/com/querydsl/sql/SelectWindowFunctionsBase.java [hunk 2]
```diff
@@ -27,7 +24,7 @@
 public class SelectWindowFunctionsBase extends AbstractBaseTest {
 
     @Test
-    @ExcludeIn(SQLSERVER) // FIXME
+    @ExcludeIn({SQLSERVER, NUODB}) // FIXME
```
Comment (timowest) [inline querydsl-sql/src/test/java/com/querydsl/sql/SelectWindowFunctionsBase.java]: Don't include this class in the NuodbSuite, if window functions are not supported

File: querydsl-sql/src/test/java/com/querydsl/sql/suites/NuoDBSuiteTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    CloseableIterator<RT> iterator = iterate(expr);
+                    return uniqueResult(iterator);
+                }
+            };
+            testQuery.addListener(new TestLoggingListener());
+            return testQuery;
+        }
+
+        @Override
+        public void Aggregate_UniqueResult() {
```
Comment (timowest) [inline querydsl-sql/src/test/java/com/querydsl/sql/suites/NuoDBSuiteTest.java]: why the override?

**Comment by timowest** [comment]
Thanks for the PR. I added some comments.

**Comment by timowest** [comment]
@tazija Could you rebase and combine the commits?

File: querydsl-sql/pom.xml [hunk 4]
```diff
@@ -275,16 +290,18 @@
           </execution>            
         </executions>
         <configuration>
-          <jdbcDriver>org.apache.derby.jdbc.EmbeddedDriver</jdbcDriver>
```
Comment (timowest) [inline querydsl-sql/pom.xml]: Could you remove the changes of this execution?

**Comment by johnktims** [comment]
@tazija Will you please rebase and make the suggested changes?

**Comment by timowest** [comment]
@tazija Can be reopened when the requested fixes are done.

**Pull request closed without merging**